### PR TITLE
support assuming role

### PR DESCRIPTION
**Pull request by cmedley**
Add Assume Role functionality #280

### CODE REVIEW

**Comment by cmedley** [comment]
Definitely ... add vcr to record assume role?

**Comment by uberblah** [comment]
not sure. the existing test infrastructure uses the stubbed versions of AWS Client objects. I'd defer to @dtan4 on that.

**Comment by cmedley** [comment]
I will do it the same way

File: spec/lib/terraforming/cli_spec.rb [hunk 1]
```diff
@@ -24,12 +28,45 @@ module Terraforming
             described_class.new.invoke(command, [], { tfstate: true, merge: tfstate_fixture_path })
           end
         end
+
+        context "with --assumes" do
```
Comment (dtan4) [inline spec/lib/terraforming/cli_spec.rb]: __[rubocop]__
* Block has too many lines. [27/25]

**Comment by DavidWittman** [comment]
This worked for me. Here's some feedback for what it's worth:

 - The help information for `--assume` should indicate that a role ARN should be passed with this option, and the README should be updated to reflect this
 - In order to assume a role from my default profile, I had to pass in `--profile default`. I expected it to assume the role from my default profile without specifying anything.

**Comment by failure84** [comment]
Working, Thanks a lot!

**Comment by dolinsky** [comment]
@cmedley was this PR intentionally closed or was it meant to be merged? I went to make use of the assumed role feature of this PR but realized it wasn't in 0.14.0 or master. I then cloned the repo locally and merged master into this branch, addressed a conflict in `lib/terraforming/version.rb`, installed the gem locally and was able to successfully assume a role in another account via 
`terraforming sg --assume arn:aws:iam::*****:role/*****`

local coverage still reports at 100% as well:
```
Finished in 4.99 seconds (files took 0.65648 seconds to load)
352 examples, 0 failures

[Coveralls] Outside the CI environment, not sending data.
Coverage report generated for RSpec to /terraforming/coverage. 2448 / 2448 LOC (100.0%) covered.
```

**Comment by cmedley** [comment]
@dolinsky I had closed it after being open a few weeks ... will reopen it

**Pull request closed without merging**